tests/ruby/misc_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+require_relative 'spec_helper'
+
+
+describe "Miscellaneous" do
+  let(:proxies) { Helpers::Pgcat.single_shard_setup("sharded_db", 5) }
+
+  describe "Clients closing connection in the middle of transaction" do
+    it "sends a rollback to the server" do
+      conn = PG::connect(proxies.main.connection_string("sharded_db", "sharding_user"))
+      conn.async_exec("SET SERVER ROLE to 'primary'")
+      conn.async_exec("BEGIN")
+      proxies.primary.begin_counting_queries
+      conn.close
+      sleep 0.5
+      # ROLLBACK, DISCARD ALL, SET application_name
+      expect(proxies.primary.end_counting_queries).to eq(3)
+    end
+  end
```

Comment:
This would be difficult to verify by looking at stats or logs of the main process because we don't increment stats for recovery `ROLLBACK`.

Putting another proxy between the database and pgcat and inspecting its logs will give us an idea what is actually being sent by the main Pgcat process